### PR TITLE
Fix the lifter for abstractconv2d.

### PR DESCRIPTION
**Pull request by abergeron**
The abstractconv tests should already test for that.

### CODE REVIEW

File: theano/sandbox/gpuarray/opt.py [hunk 1]
```diff
@@ -811,7 +811,7 @@ def local_lift_abstractconv2d(node, context_name):
                                    context_name=context_name)
     inps[1] = as_gpuarray_variable(node.inputs[1],
                                    context_name=context_name)
-    return [node.op(*inps)]
```
Comment (nouiz) [inline theano/sandbox/gpuarray/opt.py]: what was the consequence of this? I suppose an optimization warning and the op not being lifted? Is that detected by a test?

**Comment by abergeron** [comment]
I've made a different fix since the problems comes from elsewhere.

**Comment by lamblin** [comment]
I'm OK with the new fix. Is it possible to make a small test, where the shapes of a convolution gradient are coming from the GPU?

File: theano/sandbox/gpuarray/opt.py [hunk 2]
```diff
@@ -806,6 +806,9 @@ def local_error_convop(node, context_name):
             AbstractConv2d_gradWeights,
             AbstractConv2d_gradInputs])
 def local_lift_abstractconv2d(node, context_name):
+    if isinstance(node.outputs[0].type, GpuArrayType):
+        # Don't handle this node here.
```
Comment (nouiz) [inline theano/sandbox/gpuarray/opt.py]: As this need a test, can you change this comment to tell:

# the op is already on the GPU

**Pull request merged**